Imagine/Filter/PostProcessor/AbstractPostProcessor.php

```diff
@@ -0,0 +1,185 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Imagine\Filter\PostProcessor;
+
+use Liip\ImagineBundle\Binary\BinaryInterface;
+use Liip\ImagineBundle\Binary\FileBinaryInterface;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+abstract class AbstractPostProcessor implements PostProcessorInterface
+{
+    /**
+     * @var string
+     */
+    protected $executablePath;
+
+    /**
+     * @var string|null
+     */
+    protected $temporaryRootPath;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @param string      $executablePath
+     * @param string|null $temporaryRootPath
+     */
+    public function __construct($executablePath, $temporaryRootPath = null)
+    {
+        $this->executablePath = $executablePath;
+        $this->temporaryRootPath = $temporaryRootPath;
+        $this->filesystem = new Filesystem();
+    }
+
+    /**
+     * @param array $arguments
+     * @param array $options
+     *
+     * @return Process
+     */
+    protected function createProcess(array $arguments = [], array $options = []): Process
+    {
+        $process = new Process($arguments);
+
+        if (!isset($options['process'])) {
+            return $process;
+        }
+
+        if (isset($options['process']['timeout'])) {
+            $process->setTimeout($options['process']['timeout']);
+        }
+
+        if (isset($options['process']['working_directory'])) {
+            $process->setWorkingDirectory($options['process']['working_directory']);
+        }
+
+        if (isset($options['process']['environment_variables']) && is_array($options['process']['environment_variables'])) {
+            $process->setEnv($options['process']['environment_variables']);
+        }
+
+        return $process;
+    }
+
+    /**
+     * @param BinaryInterface $binary
+     *
+     * @return bool
+     */
+    protected function isBinaryTypeJpgImage(BinaryInterface $binary): bool
+    {
+        return $this->isBinaryTypeMatch($binary, ['image/jpeg', 'image/jpg']);
+    }
+
+    /**
+     * @param BinaryInterface $binary
+     *
+     * @return bool
+     */
+    protected function isBinaryTypePngImage(BinaryInterface $binary): bool
+    {
+        return $this->isBinaryTypeMatch($binary, ['image/png']);
+    }
+
+    /**
+     * @param BinaryInterface $binary
+     * @param string[]        $types
+     *
+     * @return bool
+     */
+    protected function isBinaryTypeMatch(BinaryInterface $binary, array $types): bool
+    {
+        return in_array($binary->getMimeType(), $types, true);
+    }
+
+    /**
+     * @param BinaryInterface $binary
+     * @param array           $options
+     * @param null            $prefix
+     *
+     * @return string
```

Comment:
such docblocks don't add any information. in projects i work on, i don't do them anymore unless i actually add information (e.g. string[] for an array, or explanations about the parameters). when i add them, i only add the parameters / return where i actually have something to add. otherwise its redundancy imho.

i would suggest doing the same with new code being added.

Reply:
totally agree, I'll take a look at those docblocks